ktor-http/ktor-http-cio/jvm/src/io/ktor/http/cio/CIOMultipartDataBase.kt

```diff
@@ -67,8 +68,18 @@ public class CIOMultipartDataBase(
         val contentDisposition = headers["Content-Disposition"]?.let { ContentDisposition.parse(it.toString()) }
         val filename = contentDisposition?.parameter("filename")
 
+        val body = part.body
         if (filename == null) {
-            val packet = part.body.readRemaining(formFieldLimit.toLong()) // TODO fail if limit exceeded
+            val packet = body.readRemaining(formFieldLimit.toLong())
+            if (!body.isClosedForRead) {
+                body.awaitContent()
+                if (!body.isClosedForRead) {
+                    packet.release()
+                    val cause = IllegalStateException("Form field size limit exceeded: $formFieldLimit")
+                    body.cancel(cause)
+                    throw cause
+                }
+            }
```

Comment:
Could introduce an extension method here like `body.assertClosedForRead { packet.release(); "Form field size limit exceeded: $formFieldLimit" }` to make it a little more declarative and reuseable.

Reply:
I'll make `suspend fun ByteReadChannel.exhausted(): Boolean` instead